playbooks/common/openshift-master/config.yml

```diff
@@ -409,7 +409,17 @@
 
 - name: Create services
   hosts: oo_first_master
+  vars:
+    attach_registry_volume: "{{ groups.oo_nfs_to_config | length > 0 }}"
+  pre_tasks:
+  - set_fact:
+      nfs_host: "{{ groups.oo_nfs_to_config.0 }}"
+      exports_dir: "{{ hostvars[groups.oo_nfs_to_config.0].openshift.nfs.exports_dir }}"
+      registry_storage: "{{ hostvars[groups.oo_nfs_to_config.0].openshift.nfs.registry_storage }}"
+      registry_volume: "{{ hostvars[groups.oo_nfs_to_config.0].openshift.nfs.registry_volume }}"
```

Comment:
I'm becoming a huge fan of the filter plugins...  I'm not sure if any of our filters will let you do a collect/filter using multiple variables, but it would solve a lot of these set a var, then set_fact based on that var.

That way we could use the filter then pass a list of dicts in as a value instead of using 5 separate variables.